src/PIL/ImageDraw.py

```diff
@@ -303,6 +303,12 @@ def rounded_rectangle(
             (x0, y0), (x1, y1) = xy
         else:
             x0, y0, x1, y1 = xy
+        if x1 < x0 or y1 < y0:
+            msg = (
+                "x1 must be greater than or equal to x0,"
+                " and y1 must be greater than or equal to y0"
+            )
+            raise ValueError(msg)
```

Comment:
Shall we split this into two, so the messages can be more targeted and helpful?

For example:

```suggestion
        if x1 < x0:
            msg = (
                "x1 must be greater than or equal to x0"
            )
            raise ValueError(msg)
        if y1 < y0:
            msg = (
                "y1 must be greater than or equal to y0"
            )
            raise ValueError(msg)
```

And similarly in `_imaging.c`.